### DISCUSSION TITLE
String of white squares on LCD1602

### DISCUSSION THREAD

**Discussion by dubovyi00** [probonopd/MiniDexed#256] [Q&A]
At first, thanks for this great project! When I discovered MiniDexed, there was no doubt to make DIY FM-synth, I decided not to by Volca FM

So, I tried to run MiniDexed on Pi4 Model B without any other hardware (except MIDI Keyboard), it works correctly. BUT, when I tried to run with LCD1602 (default Chinese, I suppose), results were quite sorrowful: *pic*
![image](https://user-images.githubusercontent.com/39266733/169907477-171026c1-fc5f-476c-8c4a-6f84e4106d62.png)

I checked connection scheme - it's OK. However, I did something wrong. Which wrong thing I did? 

Thx in advance for answer!

**Comment by probonopd** [comment]
What happens on a connected HDMI screen?
Can you play notes and hear sound?

**Reply by dubovyi00** [reply]
I tried connect a display with I2C - and now it works correctly. When it was connected in "standart" way, I've been able to play with my MIDI Keyboard. Don't know what's problem was with standart connection (and I personally don't care about it) but using only 4 pins is better than using 12 haha. Maybe, I was need to solder cords to display pins, not trying to connect with bare at one edge female-to-female patch cords

**Comment by zzzghhaaa** [comment]
Do you have LCD RW pin connected to GND?

**Reply by dubovyi00** [reply]
For now, I use I2C pins only to connect LCD screen

**Comment by Lt-Asmarats** [comment]
Following this thead, I also used the I2C but got the same problem, Audio and midi work perfectly.

**Reply by diyelectromusic** [reply]
That is usually a problem with the contrast on the display.  When wired directly, the resistor sets the contrast value to something vaguely sensible, but it is possible to include a pot to make it adjustable.

I think the I2C backpacks for these displays often includes a trim pot to set the contrast.

Kevin

**Reply by Lt-Asmarats** [reply]
Hello! Thanks for answering. I got the contrast all the way up and still no display other than white squares. I tried lowering the contrast but nothing. ![IMG_20240821_184412.jpg](https://github.com/user-attachments/assets/23130e94-d696-4701-9a4d-b656ee3a6688)

**Comment by Kirtai** [comment]
I had this and I *think* it turned out to be that I'd connected the I2C pins backwards.

**Reply by Lt-Asmarats** [reply]
I already checked the whole pinout and it seems okay, what do you mean by backwards exactly?

**Reply by diyelectromusic** [reply]
What is your minidexed.ini configuration?  What I2C address are you using?  At this point I'd probably try it myself with an Arduino and an I2C scanner to see what it thinks its address is :)

Kevin

**Reply by Lt-Asmarats** [reply]
I think I got to the core of the problem. I'm using an LCD Qapass 1602A screen which I think is not compatible with the program as it doesn't appear on the .ini

**Reply by diyelectromusic** [reply]
I'm not sure I follow - what do you mean "doesn't appear on the ini"?  Any 1602 display ought to work really.

And those expanders are almost always based on the PCF8574 or PCF8574a - the only difference being the range of I2C addresses they support.  IIRC the non-A version is 0x20-0x27 and the A version is 0x38-0x3F.

Kevin

**Reply by Lt-Asmarats** [reply]
Ive been investigating and I2C doesnt have any PCF inscription on the chip, it just says:
"8574T
232101"

I dont know almost anything about all of this so Im really lost tbh, thanks a lot for trying to help :DDD

Edit: Im a dumbass I just saw that it has the same serial number my bad
Edit 2: I changed the I2c Adress to 2x20 on the .ini file but doesnt work either

"LCDI2CAddress=0x20"

**Reply by diyelectromusic** [reply]
The "T" I think just relates to the package type (SO16 from the datasheet I saw) so yes, this should be a non-A variant using one of the 0x20-0x27 range of addresses.

If I can believe the schematic from here: https://alselectro.wordpress.com/2016/05/12/serial-lcd-i2c-module-pcf8574/
it looks like the address pins are pulled HIGH by default if no solder jumpers are in place, which, looking at the [PCF8574 datasheet](https://www.nxp.com/docs/en/data-sheet/PCF8574_PCF8574A.pdf) would mean the address is 0x27.

Can you post your ini file just to check everything? I think the key entries for you would be:
```
LCDEnabled=1
LCDI2CAddress=0x27
```

Kevin

**Reply by Lt-Asmarats** [reply]
I changed the adress and still nothing

The .ini file is:
```
#
# minidexed.ini
#

# Sound device
#SoundDevice=i2s
SoundDevice=pwm
#SoundDevice=hdmi
SampleRate=48000
#ChunkSize=256
DACI2CAddress=0
ChannelsSwapped=0
# Engine Type ( 1=Modern ; 2=Mark I ; 3=OPL )
EngineType=1

# MIDI
MIDIBaudRate=31250
#MIDIThru=umidi1,ttyS1
IgnoreAllNotesOff=0
MIDIAutoVoiceDumpOnPC=0
HeaderlessSysExVoices=0
# Program Change enable
#   0 = Ignore all Program Change messages.
#   1 = Respond to Program Change messages.
MIDIRXProgramChange=1
# Program Change mode
#   0 = Only recognise Program Change 0-31.
#   1 = Support 0-127 across four consecutive banks.
# NB: Only relevant if PerformanceSelectChannel=0
ExpandPCAcrossBanks=1
# Program Change action:
#   0 = Program Change messages select voices on the channel associated with each TG.
#   1-16 = Program Change messages on this channel select performances.
#   >16 = Program Change messages on ANY channel select performances.
# NB: In performance mode, all Program Change messages on other channels are ignored.
PerformanceSelectChannel=0

# HD44780 LCD
LCDEnabled=1
LCDPinEnable=17
LCDPinRegisterSelect=4
LCDPinReadWrite=0
LCDPinData4=22
LCDPinData5=23
LCDPinData6=24
LCDPinData7=25
LCDI2CAddress=0x27

# SSD1306 LCD
# For a 128x32 display, set LCDColumns=20; LCDRows=2
# For a 128x64 display, set LCDColumns=20; LCDRows=4
SSD1306LCDI2CAddress=0x0
SSD1306LCDWidth=128
SSD1306LCDHeight=32
SSD1306LCDRotate=0
SSD1306LCDMirror=0

# ST7789 LCD
# SPIBus=0 for any RPi (GPIO 10,11,8,7).
#     Note: Leave blank (default) if no SPI device required.
# Select=0|1 for CE0 or CE1
# Data = GPIO pin number
# Optional: Reset, Backlight = GPIO pin numbers
# Rotation=0,90,180,270
# SmallFont=0 (default), 1
#
# For a 240 wide display set LCDColumns=15 with LCDRows=2
SPIBus=
ST7789Enabled=0
ST7789Data=
ST7789Select=
ST7789Reset=
ST7789Backlight=
ST7789Width=240
ST7789Height=240
ST7789Rotation=0
ST7789SmallFont=0

# Default is 16x2 display (e.g. HD44780)
LCDColumns=16
LCDRows=2

# GPIO Button Navigation
#  Any buttons set to 0 will be ignored
ButtonPinPrev=0
ButtonActionPrev=
ButtonPinNext=0
ButtonActionNext=
ButtonPinBack=11
ButtonActionBack=longpress
ButtonPinSelect=11
ButtonActionSelect=click
ButtonPinHome=11
ButtonActionHome=doubleclick
ButtonPinShortcut=11
# (Shortcut doesn't have an action)

# GPIO Program/TG Selection
#  Any buttons set to 0 will be ignored
ButtonPinPgmUp=0
ButtonActionPgmUp=
ButtonPinPgmDown=0
ButtonActionPgmDown=
ButtonPinTGUp=0
ButtonActionTGUp=
ButtonPinTGDown=0
ButtonActionTGDown=

# Timeouts in milliseconds for double click and long press
DoubleClickTimeout=400
LongPressTimeout=400

# MIDI Button Navigation
#  Specify MIDI CC to act as a button (0 = ununsed, so don't use CC 0)
#  NB: Off < 64 < ON
# CC channel: 0=OFF; 1-16 MIDI Ch; >16 Omni
# If MIDIButtonNotes>0 then treat MIDIButton numbers as MIDI
# Note numbers, triggered with NoteOn/NoteOff, not CC numbers.
MIDIButtonCh=0
MIDIButtonNotes=0
MIDIButtonPrev=0
MIDIButtonNext=0
MIDIButtonBack=0
MIDIButtonSelect=0
MIDIButtonHome=0
MIDIButtonPgmUp=0
MIDIButtonPgmDown=0
MIDIButtonTGUp=0
MIDIButtonTGDown=0

# KY-040 Rotary Encoder
EncoderEnabled=1
EncoderPinClock=10
EncoderPinData=9

# Debug
MIDIDumpEnabled=0
ProfileEnabled=0

# Performance
PerformanceSelectToLoad=1
```

**Reply by diyelectromusic** [reply]
Well I'm not spotting anything obviously wrong :(

I think at this point I'd go back to first principles.  Ideally, with an Arduino I'd check the address using the I2C scanner program, although I think there is an I2C scan for Linux/RPiOS?  So you could perhaps boot the Pi into an OS and try to check the address.

Then I'd be double checking the wiring and checking the contrast works - note they are often quite sensitive and not always very smooth to change (so it might go from nothing to full quite quickly).  Do you know if the backpack is powered by 5V or 3V3 and does it include level shifters?  You can probably check the pulled-up voltage on either of the I2C lines with a multiemeter.  If they are not level shifting, then there is the possibility of have I2C lines pulled up to 5V but connected to the Pis 3V3 tolerant IO lines, which probably won't end well...

I trust there is nothing odd appearing on the HDMI log from the Pi?

After that, I'm out of ideas for the moment - sorry...

Kevin

**Reply by Lt-Asmarats** [reply]
I actually just solved it, I tried using another screen (same model but without pins) and I welded the controller to it and it worked, maybe it was a problem of the current or something
